src/server/search/doc_accessors.cc

```diff
@@ -106,13 +109,29 @@ SearchDocData StringMapAccessor::Serialize(const search::Schema& schema) const {
 struct JsonAccessor::JsonPathContainer : public jsoncons::jsonpath::jsonpath_expression<JsonType> {
 };
 
-string_view JsonAccessor::GetString(string_view active_field) const {
-  auto res = GetPath(active_field)->evaluate(json_);
-  DCHECK(res.is_array());
-  if (res.empty())
-    return "";
-  buf_ = res[0].as_string();
-  return buf_;
+BaseAccessor::StringList JsonAccessor::GetString(string_view active_field) const {
+  auto path_res = GetPath(active_field)->evaluate(json_);
+  DCHECK(path_res.is_array());  // json path always returns arrays
+
+  if (path_res.empty())
+    return {};
+
+  if (path_res.size() == 1) {
+    buf_ = path_res[0].as_string();
+    return {buf_};
+  }
+
+  buf_.resize(sizeof(buf_) + 1);  // prevent sso
+  buf_.clear();
+
+  StringList out;
+  for (auto element : path_res.array_range()) {
+    size_t start = buf_.size();
+    buf_ += element.as_string();
```

Comment:
This could invalidate all previous `string_view`s in `out`, no?
Is there anything in line 124 that prevents growing again?

Reply:
No 😆 I forgot to use my brain
<img src="https://github.com/dragonflydb/dragonfly/assets/20553775/984c4c1a-e5ab-44cf-aac5-7478a145ef0a" height="150" />